lib/reservation.rb

```diff
@@ -0,0 +1,47 @@
+require "date"
+require "time"
+
+module Hotel
+  class Reservation
+    attr_reader :booking_ref
+    attr_accessor :booking_ref, :number_of_rooms, :first_room_number, :check_in, :check_out, :total_cost, :room_blocks
+
+    RATE = 200.00
+    DISCOUNTED_RATE = 160.00
+
+    def initialize(booking_ref:, number_of_rooms:, first_room_number:, check_in:, check_out:, total_cost:, room_blocks:)
+      @booking_ref = booking_ref
+
+      if (number_of_rooms <= 5)
+        @number_of_rooms = number_of_rooms
```

Comment:
There is no requirement that a reservation should have more than one room. Blocks should have different behavior: they are not a type of reservation